### PR TITLE
feature(Management List): A new design pattern for sorting/filtering items #260

### PR DESCRIPTION
**Pull request by richavyas**
## Description

Management List, a new design pattern for sorting/filtering items. Takes care of #260 

### What's included?
- Management List with sorting & filtering design pattern. 
- Filter options to be added as one of the column headers and the list can be filtered based on what option has been selected. 
- Based on the selected filter option, list can also be sorted say `asc/desc`.

#### Test Steps

- [x] ng serve
- [x] then navigate to http://localhost:4200/#/style-guide/management-list
- [x] 🎉 

filter-sort-menu

### CODE REVIEW

**Comment by kyleledbetter** [comment]
I'm not sure if design patterns and non-platform code needs unit tests

File: src/platform/core/management-list/management-list.module.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    CovalentCommonModule.forRoot(),
+  ],
+  declarations: [
+    TdManagementListComponent,
+  ],
+  exports: [
+    TdManagementListComponent,
+  ],
+})
+export class CovalentManagementListModule {
```
Comment (emoralesb05) [inline src/platform/core/management-list/management-list.module.ts]: Im not sure we want to add a module just for a design pattern. The whole code should be in the design pattern page.

File: src/platform/core/management-list/management-list.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  [key: string]: OrderBy;
+}
+
+@Component({
+  selector: 'td-management-list',
+  templateUrl: './management-list.html',
+  styleUrls: ['./management-list.scss'],
+})
+
+export class TdManagementListComponent {
```
Comment (emoralesb05) [inline src/platform/core/management-list/management-list.ts]: Any reason this is a component and not a service?

File: src/app/components/style-guide/management-list/management-list.component.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  <md-card-content>
+    <div class="md-padding bgc-grey-200">
+      <md-card>
+        <md-list class="pull-top">
+          <md-list-item>
+            <md-icon md-list-avatar>apps</md-icon>
+            <h3 md-line>App Name</h3>
+            <span flex></span>
+            <span flex="35" class="pad-left md-body-1 tc-grey-500">Owner</span>
+            <span class="md-sort-header" flex="40" layout="row" layout-align="center center" class="pad-left md-body-1 tc-grey-500">
```
Comment (emoralesb05) [inline src/app/components/style-guide/management-list/management-list.component.html]: We can probably leverage the `md-select` now to do this.

File: src/app/components/style-guide/management-list/management-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  dateOptions: Object[] = ['Updated', 'Created'];
+  dateFilter: string = 'Updated';
+  sortKey: string = 'updated_at';
+  headers: IHeaders = {
+    updated_at: OrderBy.ASC,
+    created_at: OrderBy.ASC,
+  };
+
+  constructor(@Inject(forwardRef(() => TdManagementListComponent)) private _parent: TdManagementListComponent) {
```
Comment (emoralesb05) [inline src/app/components/style-guide/management-list/management-list.component.ts]: For a `design pattern` purpose, i think its better to just have the `sortBy` function here.

I get why we would want to reuse it, but we always want to push people to sort in the server, and not in the client.

**Comment by kyleledbetter** [comment]
@richavyas this looks fantastic. As discussed let's keep the pattern to just mock & pattern, and update the spec in the issue https://github.com/Teradata/covalent/issues/260

So i'd recommend moving any logic the platform for now and keep this limited to design pattern in the docs app

**Comment by richavyas** [comment]
- [x] Changed this to pattern & mock
- [x] replaced menu to `select` to load filter options

filter-sort-select

**Comment by kyleledbetter** [comment]
Pushed updates:
manage-list-pattern

**Comment by emoralesb05** [comment]
Pushed a couple of tweaks to leverage the usage of `md-select` and make the example more generic

**Pull request merged**